src/transformers/models/t5/modeling_flax_t5.py

```diff
@@ -981,7 +987,10 @@ def __call__(
 
         # prepare encoder inputs
         if attention_mask is None:
-            attention_mask = jnp.ones_like(input_ids)
+            if input_ids is not None:
```

Comment:
Instead of a nested set of `if`/`else` statements, could we first establish the batch-size and seq-len from either the `input_ids` or `inputs_embeds`:
```python
if input_embeds is None:
    batch_size, sequence_length = input_ids.shape
else:
    batch_size, sequence_length = input_embeds.shape[:-1]
```
And then feed these into the attention mask:
```python
if attention_mask is None:
    attention_mask = jnp.ones(batch_size, sequence_length)
```
This sort of style helps a lot with readability! In general with Transformers, we advocate for longer, more readable code than concise, condensed code.